crates/bevy_dev_tools/src/fps_overlay.rs

```diff
@@ -26,19 +29,31 @@ pub struct FpsOverlayPlugin {
     pub config: FpsOverlayConfig,
 }
 
+impl FpsOverlayPlugin {
+    /// [`DevToolId`] of the `FpsOverlayPlugin`.
+    pub const FPS_OVERLAY_ID: DevToolId =
+        DevToolId::from_u128(66095722480681667677084752066997432964);
+}
+
 impl Plugin for FpsOverlayPlugin {
     fn build(&self, app: &mut bevy_app::App) {
         // TODO: Use plugin dependencies, see https://github.com/bevyengine/bevy/issues/69
         if !app.is_plugin_added::<FrameTimeDiagnosticsPlugin>() {
             app.add_plugins(FrameTimeDiagnosticsPlugin);
         }
         app.insert_resource(self.config.clone())
+            .register_dev_tool(DevTool::new(Self::FPS_OVERLAY_ID))
             .add_systems(Startup, setup)
             .add_systems(
                 Update,
                 (
                     customize_text.run_if(resource_changed::<FpsOverlayConfig>),
-                    update_text,
+                    change_visibility.run_if(resource_changed::<DevToolsStore>),
```

Comment:
I don't like this. This will run every time any tool has been changed, but I can't think of a solution to only run if a specific tool has been changed

Reply:
I think we can relax on that somehow by just making a global system that manages this, based on one-shot systems, I'm not sure how effective this can be do though. At the same time, that would be useless without some kind o cache to store what was changed and what was not.

Reply:
I think we can relax on that somehow by just making a global system that manages this, based on one-shot systems, I'm not sure how effective this can be do though. At the same time, that would be useless without some kind o cache to store what was changed and what was not.

Reply:
If we had some kind of cache storing what was changed, I don't think there would be any reason for complicating it with the global system. I'd just let the user interact with the cache and use it in a `run_if` condition.

Reply:
I think this is a fine level of granularity for now: I wouldn't expect `DevToolsStore` to change regularly at all.